database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostListViewModel.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  Copyright (c) 2021 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import Firebase
+
+class PostListViewModel: ObservableObject {
+  @Published var isLoading = false
+  @Published var alert = false
+  @Published var alertMessage = ""
+  @Published var posts: [PostViewModel] = []
+
+  // setup instance of FIRDatabaseReference for reading and writing data
+  private var ref = Database.root
+  private var refHandle: DatabaseHandle?
+
+  // function called when error occurs
+  func showAlertMessage(message: String) {
+    alertMessage = message
+    alert.toggle()
+  }
+
+  func getCurrentUserID() -> String? {
```

Comment:
minor: If the method is not used outside of the class it should be marked as private. The same is applicable to other methods/classes.